src/command_modules/azure-cli-redis/azure/cli/command_modules/redis/_params.py

```diff
@@ -3,43 +3,64 @@
+        # pylint: disable=line-too-long
+        c.argument('zones', type=shell_safe_json_parse, help='A list of availability zones denoting where the resource needs to come from')
         c.argument('shard_id', type=int)
         c.argument('sku', help='Type of Redis cache.', arg_type=get_enum_type(SkuName))
-        c.argument('vm_size', help='Size of Redis cache to deploy. Example : values for C family (C0, C1, C2, C3, C4, '
-                                   'C5, C6). For P family (P1, P2, P3, P4)')
+        # pylint: disable=line-too-long
+        c.argument('minimum_tls_version', help='Specifies the TLS version required by clients to connect to cache', arg_type=get_enum_type(TlsVersion))
+        # pylint: disable=line-too-long
+        c.argument('vm_size', arg_type=get_enum_type(allowed_c_family_sizes + allowed_p_family_sizes), help='Size of Redis cache to deploy. Basic and Standard Cache sizes start with C. Premium Cache sizes start with P')
         c.argument('enable_non_ssl_port', action='store_true')
-        c.argument('shard_count', type=int)
-        c.argument('tags', tags_type)
-
-    for scope in ['redis import-method', 'redis import']:
-        with self.argument_context(scope) as c:
-            c.argument('files', nargs='+', help='Space-separated list of files to import.')
-            c.argument('file_format', help='File format to import.')
 
-    with self.argument_context('redis patch-schedule set') as c:
-        c.argument('schedule_entries', type=ScheduleEntryList)
+    with self.argument_context('redis firewall-rules list') as c:
+        c.argument('cache_name', arg_type=cache_name_without_id_part)
+        c.argument('rule_name', help='Name of the firewall rule')
 
-    with self.argument_context('redis create') as c:
-        c.argument('name', arg_type=name_type, completer=None)
-        c.argument('tenant_settings', type=JsonString)
+    with self.argument_context('redis server-link') as c:
+        c.argument('name', arg_type=cache_name_without_id_part)
+        c.argument('server_to_link', help='Resource ID or name of the redis cache to be linked')
+        c.argument('replication_role', help='Role of the redis cache to be linked', arg_type=get_enum_type(ReplicationRole))
+        c.argument('linked_server_name', help='Name of the linked redis cache')
 
-    with self.argument_context('redis export') as c:
-        c.argument('container', help='Container name to export to.')
-        c.argument('prefix', help='Prefix to use for exported files.')
-        c.argument('file_format', help='File format to export.')
+    with self.argument_context('redis patch-schedule') as c:
+        c.argument('name', arg_type=cache_name_without_id_part)
+        # pylint: disable=line-too-long
+        c.argument('schedule_entries', help="List of Patch schedule entries. Example Value:[{\"dayOfWeek\":\"Monday\",\"startHourUtc\":\"00\",\"maintenanceWindow\":\"PT5H\"}]", type=ScheduleEntryList)
```

Comment:
This example shows why JSON strings translate to painful UX. Feel free to set up a time to go over options on improving this.

Reply:
Since this command is already out, with it accepting JSON string as parameter, i think it's okay to leave as is in the scope of this review. You can file an issue against me, and i will fix it with in the next release.